include/revng/ABI/Definition.h

```diff
@@ -103,6 +103,21 @@ type: struct
       (if `ArgumentsArePositionBased` is `false`).
     type: bool
 
+  - name: NoRegisterArgumentsCanComeAfterStackOnes
+    doc: |
+      States whether ABI allows a stack argument (mainly one too big
+      to be placed in the registers) to precede other register arguments.
+
+      For example, if there is a `Big` struct that has to use the stack, and
+      a function like
+      ```
+      void function(Big, uint32_t);
+      ```
+      if this value is set to true, both argument will be passed on stack,
+      otherwise, only the struct will.
+    type: bool
+    optional: true
+
```

Comment:
A new abi::Definition option:
https://github.com/revng/revng/pull/232/commits/3ce4f394acc7bb0fa84c15ca2cdcf19b72ecbfaa

@aleclearmind, feel free to close this if the commit looks good.

Reply:
`ABI: introduce an additional definition option`

Reply:
Is this somehow related to positional vs non-positional ABIs? Could this fall under that umbrella or maybe you just see a benefit to have this on its own?

Reply:
Positional ABIs cannot have this be any relevant by definition because they ALWAYS have the first X  arguments in registers (either as a value, or as a pointer-to-copy) - so arguments starting from X+1 are always on stack - hence there is no ambiguity.

This is specifically for differentiating ABIs like `SystemV_x64_86` which lets passing `pointer` argument in registers as opposed to `ARM` which does not:
```cpp
void whatever(struct big_object_on_stack, void pointer);
```